Source/ARTWebSocketTransport.m

```diff
@@ -194,6 +195,22 @@ - (NSURL *)setupWebSocket:(__GENERIC(NSArray, NSURLQueryItem *) *)params withOpt
     NSURLQueryItem *libParam = [NSURLQueryItem queryItemWithName:@"lib" value:[ARTDefault libraryVersion]];
     queryItems = [queryItems arrayByAddingObject:libParam];
 
+    // Transport Params
+    if (options.transportParams != nil) {
+        /**
+         Remove the same, previously  set values aka overwrite previous values for the same keys
+         */
+        NSPredicate *predicateKeyName = [NSPredicate predicateWithFormat:@"name IN %@", [options.transportParams allKeys]];
+        NSMutableArray *mutableQueryItems = [queryItems mutableCopy];
+        NSArray *filteredObjects = [queryItems filteredArrayUsingPredicate:predicateKeyName];
+        [mutableQueryItems removeObjectsInArray:filteredObjects];
+        queryItems = mutableQueryItems;
```

Comment:
NSArray is better because it stores just `NSURLQueryItem` that has `name` and `value` properties, so there is no need to create another key in NSDictionary since `NSURLQueryItem` has one.

Reply:
I think original implementation of this function doesn't take into account that params from one source can override another. So this addition looks more like a trick on a bad implementation. Better to rewrite it with dictionaries. IMHO (not fatal either).

Reply:
here a852bed7559cc4935472e7ea664549b3d9c7ba66

Reply:
I am happy to resolve this if @maratal is okay as well

Reply:
@ben-xD this one is OUTDATED now :) - could you resolve this conversation or we are waiting for something else?